### ISSUE TITLE
Calling `q::debug` with a single argument should work

### ISSUE THREAD

**Issue by bugarela** [informalsystems/quint#1567]
Currently, if I want to add debug statements to an expression like:
```bluespec
foo(bar(1 + 2))
```

I need to do something like:
```bluespec
q::debug("foo", foo(q::debug("bar", bar(q::debug("plus", 1 + 2))))))
```

I'd like to also be able to write:
```bluespec
q::debug(foo(q::debug(bar(q::debug(1 + 2))))))
```

And have Quint print the expressions:
```bluespec
> foo(bar(1 + 2)): 45
> bar(1 + 2): 10
> 1 + 2: 3
```

Moreover (not this issue), we can also consider having something like:
```bluespec
q::deepDebug(foo(bar(1 + 2)))
```

**Comment by halcyonet** [comment]
@bugarela Can I try to solve this?

**Comment by bugarela** [comment]
> [@bugarela](https://github.com/bugarela) Can I try to solve this?

@halcyonet of course, you are more than welcomed! Feel free to ask for help if needed

**Comment by halcyonet** [comment]
Hi @bugarela . Hope you are having a good day. I am thinking of implementing a macro-like transformation approach where:
I will add logic in ToIrListener.ts to transform any single-argument the source text of the expression. This would happen during parsing rather than runtime, so we wouldn't need to modify the Rust/TypeScript runtime implementations. Basically I would modify the pushApplication method in ToIrListener.ts to detect q::debug calls with one argumentExtract the source text of the expression from the parser context. Does this approach seems appropriate? I thought it would be cleaner than modifying the runtime functions. 
Please let me know or you could suggest any other better approach to solve this issue.

**Comment by bugarela** [comment]
That seems like a great idea, as it will also prevent it from triggering type errors (as the type checker runs before the runtime, and would complain if there is only one argument). I initially though of doing it in the runtime, but your way should be better in all the aspects I can think of 😄

**Comment by halcyonet** [comment]
Hi @bugarela . I have send a PR #1660 . Could you kindly review and let me know?
